### PR TITLE
Fix problem when running config-file tests

### PR DESCRIPTION
**Pull request by castulo**
The commit also fixes a bug with the same tests which was causing the
test environment to be left behind.

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

### CODE REVIEW

File: test/functional/testlib.bash [hunk 1]
```diff
@@ -1780,22 +1778,47 @@ destroy_test_fs() { # swupd_function
 
 create_config_file() { # swupd_function
 
+	SWUPD_CONFIG_DIR="$("$SWUPD" --version | grep "config file path" | awk '{print $4}')"
```
Comment (otaviobp) [inline test/functional/testlib.bash]: This isn't going to work for paths separated with ':'. like the default path.

File: test/functional/testlib.bash [hunk 2]
```diff
@@ -1780,22 +1778,47 @@ destroy_test_fs() { # swupd_function
...
+
+	# when the config dir is set to a relative path, lets redirect it to the
+	# directory of the test environment so every test has its own config file
+	# this is useful when running the config tests in parallel with other tests,
+	# if the path is absolute lets leave it as it is
+	if [ "$SWUPD_CONFIG_DIR" == "./testconfig" ]; then
```
Comment (otaviobp) [inline test/functional/testlib.bash]: Why do we need to have this case supported? And why it should be specific for ./testconfig/

File: test/functional/testlib.bash (hunk 1 above)
Comment (otaviobp) [reply test/functional/testlib.bash]: And I'm a bit concerned on overwriting any possible config file that the user may have.

Comment (castulo) [reply test/functional/testlib.bash]: yes, that is right, this won't work if the user has more than one path defined.
And about the possibility of overwriting an config file a user may have,  that should not happen , that is why I'm backing it up first if config file already exists.

File: test/functional/testlib.bash (hunk 2 above)
Comment (castulo) [reply test/functional/testlib.bash]: well, currently if we want to run the tests in parallel, we need to create one config file per test, which forces us to change the directory where swupd reads the config file from. So if the user wants to run tests in parallel he/she will need to configure the build with ` --with-config-file-path=./testconfig` like we are doing right now. That is why I left this particular condition here, I don't see how else we could support multiple config files without having to add a flag for setting up the config file. I'm open to suggestions if you have any.

**Comment by castulo** [comment]
V2:

- Removed the code that looked for an absolute path in the config file
- Skip the config tests if the env is not configured correctly to run the tests
- Make sure the working directory is returned to the original so the test environment gets removed correctly

**Pull request merged**